docs/bazel/status.md

```diff
@@ -0,0 +1,339 @@
+# Coda Bazelization Project
+
+WARNING: this document is outdated by a few weeks.  Check back in a few days.
+
+# Table of Contents
+* [Goals and Non-goals](#goals)
+* [Tasks](#tasks)
+  * [C/C++](#tasks_c)
+  * [OCaml](#tasks_ocaml)
+  * [Go](#tasks_go)
+  * [Rust](#tasks_rust)
+  * [Javascript](*tasks_js)
+  * [Integration](#tasks_integration)
+* [Status](#status)
+* [Schedule](#schedule)
+* [Working with Bazel](#workingwithbazel)
+* [Resources](#resources)
+
+## Goals and Non-goals <a name="goals"></a>
+
+Mandatory Goals:
+
+* Complete Bazel coverage of the Coda codebase:
+  * OCaml code
+  * C/C++ code and dependencies.
+  * Golang dependencies ([go-libp2p](https://github.com/libp2p/go-libp2p))
+  * Rust dependencies ([marlin](https://github.com/scipr-lab/marlin)?)
+  * Reason/Javascript
+* "Just works" on:
+  * Linux
+  * MacOS
+
+Non-mandatory goals:
+
+* Cross-platform builds targeting Linux, x86_64
+  * In particular, MacOS -> Linux
+* Dockerfile to support Linux development of Coda on Mac host
+
+Non-goals:
+
+* Windows support.  Maybe later.
+* Cross-platform build targeting ARM.
+* Production-quality `rules_ocaml` Bazel library.  The goal is
+  implement just enough functionality to build Coda.  Supporting a
+  production-quality general-purpose library of OCaml rules for Bazel
+  is out of scope.
+
+## Tasks <a name="tasks"></a>
+
+General:
+
+* Customization/Optimiztion configurations.
+  * User-friendly support for settings e.g. debug/optimized build, etc.
+  * Support for dependency configuration.  For example, libsnark
+    supports various curves (bn128, etc.), building with or without
```

Comment:
I haven't looked at this code for a while, but I think it will do what you want.  That is, a build-time configuration setting (perhaps controlled by a profile flag) will control which curve(s) get built, and with which params.  Is that what you mean?

Reply:
If there's an easy way to say 'this OCaml library links against these curves' and have it link against a library built with those curves, then this meets my requirements. I don't want to have to specify the curves to build for an entire build, since it's an implementation detail of those specific OCaml modules.

Reply:
That should be doable.